harness/features/variable.local.test.ts

```diff
@@ -0,0 +1,356 @@
+
+            beforeAll(async () => {
+                scope
+                    .get(configRequestUrl)
+                    .delay(2000)
+                    .reply(404)
+
+                // create client          
+                await createClient(url, uninitClientId, 'dvc_server_test_token', {
+                    baseURLOverride: uninitBaseURL,
+                    eventFlushIntervalMS: 500,
+                    configPollingIntervalMS: 100000
+                })
+                await waitForRequest(
+                    scope,
+                    configRequestUrl,
+                    3000,
+                    'Config request timed out'
+                )
+            })
+
+            forEachVariableType((type) => {
+                const { key, defaultValue } = expectedVariablesByType[type]
+
+                it('should return default value if client is uninitialized',  async () => {
+                    const variableResponse = await callVariable(
+                        uninitClientId, url, variationOnUser.location, key, defaultValue
+                    )
+                    const variable = await variableResponse.json()
+                    expectDefaultValue(variable, defaultValue)                 
+                })
+
+                it.failing('should throw exception if user is invalid',  async () => {    
+                    const variableResponse =
+                        await callVariable( uninitClientId, url, invalidUser.location, key, defaultValue)
+                    const variable = await variableResponse.json()
+
+                    expect(variable.exception).toBe('Must have a user_id set on the user')
+                })
+            })
+        })
+    })
+
+    const expectEventBody = (
+        body: Record<string, unknown>, 
+        variableId: string, 
+        eventType: string,
+        value?: number
+    ) => {
+        expect(body).toMatchObject({
+            batch: [{ 
+                user: expect.objectContaining({
+                    platform: 'NodeJS',
+                    sdkType: 'server',
+                }),
+                events: [
+                    expect.objectContaining({
+                        type: eventType,
+                        target: variableId,
+                        value: value !== undefined ? value : 1,
+                    })
+                ]
+            }]
+        })
+    }
+
+    type ValueTypes = string | boolean | number | JSON
+
+    type VariableResponse = {
+        entityType: string,
+        data: {
+            value: ValueTypes,
+            isDefaulted: boolean,
+            defaultValue: ValueTypes
+        }
+    }
+    
+    const expectDefaultValue = (variable: VariableResponse, defaultValue: ValueTypes) => {
+        expect(variable).toEqual(expect.objectContaining({
+            entityType: 'Variable',
+            data: expect.objectContaining({
```

Comment:
can this just be an exact match? currently this allows additional fields to be present on the variable, and we probably want to test the exact set of fields we're expecting

Reply:
fixed!